src/reporters.ts

```diff
@@ -0,0 +1,160 @@
+import { type CCReport, type CCEntry, type Tree, RadonType, RadonRank } from "./types";
+import * as path from "path"
+
+const PATH_ENDS: Array<string> = [
+    ".",
+    "/"
+]
+
+// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types -- I am mutating an argument
+function TreeMerge<T>(first: Tree<T>, second: Readonly<Tree<T>>):void{
+    for (const KEY in second){
+        if (KEY in first){
+            TreeMerge(first[KEY] as Tree<T>, second[KEY] as Tree<T>);
+        } else {
+            first[KEY] = second[KEY]
+        }
+    }
+}
+
+// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types -- Almost impossible to make readonly and still being useful.
+function ToTree (data: Readonly<CCReport>) : Array<Tree<Array<CCEntry>>> {
+    const TREES: Array<Tree<Array<CCEntry>>> = [];
+    const ROOT_MAP: Record<string,Tree<Array<CCEntry>>> = {}
+    for (const F_NAME in data){
+        let tree: Tree<Array<CCEntry>> = {};
+        let temp_tree: Tree<Array<CCEntry>> = {};
+
+        tree[path.basename(F_NAME)] = data[F_NAME]
+
+        let current_name:string = path.dirname(F_NAME);
+
+        while (!PATH_ENDS.includes(current_name)){
+            temp_tree[path.basename(current_name)] = tree;
+            tree = temp_tree;
+            temp_tree = {};
+            current_name = path.dirname(current_name)
```

Comment:
I think I miss something, because I don't understand the utility of this line ?
If it no longer useful, we could change `current_name` as constant.

```suggestion
            current_name = path.dirname(current_name)
```

Reply:
This line updates the value of `current_name`. Removing this line would make the while loop infinite. I redirect you to the [node documentation about paths](https://nodejs.org/api/path.html) for further information about how the variable is updated. This loop reads all the directories referenced in the path from right to left as I create a tree in a bottom-up way.